### PR TITLE
[foundation] RelWithDebInfo should build the same as Release.

### PR DESCRIPTION
**Pull request by drodriguez**
Enable optimizations also for RelWithDebInfo builds.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,7 +59,7 @@ endif()
 add_dependencies(uuid CoreFoundation)
 
 set(swift_optimization_flags)
-if(CMAKE_BUILD_TYPE MATCHES Release)
```
Comment (compnerd) [inline CMakeLists.txt]: Oh, interesting!  This makes a lot of sense.  Please change this to:

```
if(NOT CMAKE_BUILD_TYPE MATCHES Debug)
```

Please don't merge this, you will break the Windows build.

**Comment by drodriguez** [comment]
Switch the logic as the feedback proposed. I will hold for merging.

**Comment by compnerd** [comment]
Thanks for holding off on the change.  I've tracked down the issue, and hope to get that issue dealt with in a day or so.  The problem is in LLVM, I've found the issue and have a fix for it.  Just committed it upstream and waiting on the bots to run through before I create the PR for it to pull it into stable.

**Comment by millenomi** [comment]
@compnerd Ping me on this PR when it's time to merge.

**Comment by compnerd** [comment]
Okay, at this point, this should not regress the Windows build as the compiler side of the issue has been taken care of.  Lets get it through the bots first though.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@millenomi - ping :-)

**Comment by millenomi** [comment]
@compnerd Pong.

**Pull request merged**